### PR TITLE
Add --songuri and --listuri to enable playing a song or playlist Uri

### LINKED ISSUE #5: Play Playlist URI - feature request
Are you willing to create a feature to play a specified playlist instead of just one particular song? Or is this already possible with existing commands? Thanks!

### PR DESCRIPTION
**Pull request by Myoldmopar**
Addresses:
 - Fixes #5
 - Potentially the start of addressing #15 by combining with some other service to get URI by artist/track and such, but that may be a stretch

With these new arguments, the user can provide a song or playlist URI to the CLI and it should play it.  The URI can be found by "sharing" a track from the Spotify app.  The URL will look like: 

```
https://open.spotify.com/track/0z5KIZSt0Vcm8diVg0WBjP?si=16c79071b0134c83
                               ^^^^^^^^^^^^^^^^^^^^^^
                                ^ Spotify URI Here ^
```

Maybe there's an easier way?  Anyway you can call it from the CLI like:

```
spotifycli --songuri 0z5KIZSt0Vcm8diVg0WBjP
```

or

```
spotifycli --listuri 37i9dQZF1DZ06evO0jO79m
```

And it plays just fine.  I'm totally open to adjusting the naming of the command line arguments as you prefer.

Note, this will have minor conflicts with if #85 is merged.  I'm happy to resolve them, just tell me what method you prefer for the squash/merge/rebase whatever and I can do it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ use one of the following parameters:
 --playpause       plays or pauses the song (toggles a state)
 --next            plays the next song
 --prev            plays the previous song
+--songuri OPENURI plays the track at the provided Uri
+--listuri OPENURI plays the playlist at the provided Uri
```
Comment (Myoldmopar) [inline README.md:71]: Updated the readme based on the new args

File: spotifycli/spotifycli.py [hunk 2]
```diff
@@ -58,7 +58,10 @@ def main():
         perform_spotify_action("Next")
     elif args.prev:
         perform_spotify_action("Previous")
-
+    elif args.songuri:
+        perform_spotify_action("OpenUri", f"string:spotify:track:{args.songuri}")
+    elif args.listuri:
+        perform_spotify_action("OpenUri", f"string:spotify:playlist:{args.listuri}")
```
Comment (Myoldmopar) [inline spotifycli/spotifycli.py:64]: If songuri or listuri are specified, they will come in storing the argument value, so pass this to the expanded `perform_spotify_action` function.

File: spotifycli/spotifycli.py [hunk 3]
```diff
@@ -82,32 +85,37 @@ def start_shell():
 def add_arguments():
     parser = argparse.ArgumentParser(description=__doc__)
     for argument in get_arguments():
-        parser.add_argument(argument[0], help=argument[1], action="store_true")
+        if not argument[2]:
+            parser.add_argument(argument[0], help=argument[1], action="store_true")
+        else:
+            parser.add_argument(argument[0], help=argument[1], action="store")
```
Comment (Myoldmopar) [inline spotifycli/spotifycli.py:91]: The `get_arguments` function row returns a list of `tuple[str, str, bool]` where the third argument specifies whether this is a `store` argument or a `store_true` argument.

File: spotifycli/spotifycli.py [hunk 4]
```diff
@@ -82,32 +85,37 @@ def start_shell():
...
+        ("--arturl", "shows album image url", False),
+        ("--playbackstatus", "shows playback status", False),
+        ("--play", "plays the song", False),
+        ("--pause", "pauses the song", False),
+        ("--playpause", "plays or pauses the song (toggles a state)", False),
+        ("--lyrics", "shows the lyrics for the song", False),
+        ("--next", "plays the next song", False),
+        ("--prev", "plays the previous song", False),
+        ("--songuri", "plays the track at the provided Uri", True),
+        ("--listuri", "plays the playlist at the provided Uri", True),
```
Comment (Myoldmopar) [inline spotifycli/spotifycli.py:118]: Added a third argument to each of the entries in the list which specified whether the arg will specify a value or not.  All the existing entries here should just continue to trigger a `true` value to be stored.  But in the case of the two new ones, the user will be expected to provide a value after the argument.  So those have a `True` in the argument.

I could have done this many other ways, but this seems the most intentional.  I'm certainly open to your thoughts.

File: spotifycli/spotifycli.py [hunk 5]
```diff
@@ -253,11 +261,18 @@ def get_spotify_property(spotify_property):
...
+def perform_spotify_action(spotify_command, extra_arg = None):
+    command_list = [
+        "dbus-send",
+        "--print-reply",
+        f"--dest=org.mpris.MediaPlayer2.{client}",
+        "/org/mpris/MediaPlayer2",
+        f"org.mpris.MediaPlayer2.Player.{spotify_command}",
+    ]
+    if extra_arg is not None:
+        command_list.append(extra_arg)
```
Comment (Myoldmopar) [inline spotifycli/spotifycli.py:273]: I broke out the commands into a list of strings, and append the extra arg if it is present.

File: spotifycli/spotifycli.py [hunk 6]
```diff
@@ -253,11 +261,18 @@ def get_spotify_property(spotify_property):
...
+    command_string = " ".join(command_list)  # could avoid this by taking out shell=False below
+    Popen(command_string, shell=True, stdout=PIPE)
```
Comment (Myoldmopar) [inline spotifycli/spotifycli.py:275]: I then put the list back into a string command line and let `Popen` parse it with `shell=True`.  I think it's also reasonable to say `shell=False` and pass in the nice list, but I didn't want to overstep here.  And this works totally fine anyway.

**Review by Myoldmopar — COMMENTED** [review]
Code walkthrough completed.  Looking forward to hearing your thoughts!

**Review by pwittchen — APPROVED** [review]

**Comment by pwittchen** [comment]
Thanks for the PR!

**Pull request merged**